### ISSUE TITLE
Confirm behavior of symbol-placement: point and LineStrings

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-js#3808]
Currently, a symbol layer using `symbol-placement: point` renders a symbol for only the first coordinate pair in a LineString (see e.g. [this test case](https://github.com/mapbox/mapbox-gl-test-suite/tree/master/render-tests/symbol-geometry/linestring)).

Is this the intended behavior? It would seem more logical to render a symbol for each point in the LineString (or MultiLineString).

**Comment by lucaswoj** [comment]
This behaviour has not been explicitly specified or tested until recently. 

Consistency between symbol placement and circle placement is desirable. This would be most easily acheived by adding a new `symbol-placement: centroid` mode.

**Comment by jfirebaugh** [comment]
I agree that this extends to `circle` layers as well, and the two should have equivalent placement behavior.

Can you say more about what the behavior of `symbol-placement: centroid` would be, and how it relates to the OP here (symbol-placement: point with LineStrings)?

**Comment by lucaswoj** [comment]
Quick table to clarify my point about centroids:

|                     | circle                 | symbol w/ point placement |
|---------------------|------------------------|---------------------------|
| Point geometry      | placed at vertex       | placed at vertex          |
| LineString geometry | placed at every vertex | placed at first vertex    |
| Polygon geometry    | placed at every vertex | placed at centroid        |

To get the placement of circles to match the placement of symbols in point placement mode, we'd need to replace the "placed at centroid" behaviour with "placed at every vertex" behaviour. 

I'd advocate for keeping the "placed at centroid" behaviour available under a `symbol-placement: centroid` mode.

**Comment by mourner** [comment]
@lucaswoj an alternative suggestion from me is to have `point` mean `centroid` for all types. This means:

|                     | circle                 | symbol w/ point placement |
|---------------------|------------------------|---------------------------|
| **Point geometry**      | placed at vertex       | placed at vertex          |
| **LineString geometry** | placed at line centroid | placed at line centroid    |
| **Polygon geometry**    | placed at polygon centroid | placed at polygon centroid    |

I think this is more intuitive and generally more useful. For the previous behavior, I'd expect people to use different geometry types (e.g. multipoint).

**Comment by jfirebaugh** [comment]
By centroid, do you mean "pole of inaccessibility"? My proposal would be either of these two:

|                     | circle, or symbol w/ point placement |
|---------------------|------------------------|
| **Point geometry**      | placed at vertex       |
| **LineString geometry** | placed at midpoint |
| **Polygon geometry**    | placed at PoI |

or

|                     | circle, or symbol w/ point placement |
|---------------------|------------------------|
| **Point geometry**      | placed at vertex       |
| **LineString geometry** | placed at each vertex |
| **Polygon geometry**    | placed at PoI |

**Comment by lucaswoj** [comment]
> By centroid, do you mean "pole of inaccessibility"?

Yup. I've been incorrectly using the term centroid. ("pole of inaccessibility" is so long and "POI" is so overloaded 😬)

> My proposal would be either of these two

Those both look good. 

@ansis Do you remember the motivation behind https://github.com/mapbox/mapbox-gl-js/pull/2027? Curious if there are any important use cases for the current circle behaviour.
